code/modules/projectiles/ammunition/ballistic/revolver.dm

```diff
@@ -56,3 +56,16 @@
 	name = ".38 Iceblox bullet casing"
 	desc = "A .38 Iceblox bullet casing."
 	projectile_type = /obj/projectile/bullet/c38/iceblox
+
+/obj/item/ammo_casing/pea
+	name = "pea bullet casing"
+	desc = "A pea bullet casing. Yummy."
+	caliber = CALIBER_PEA
+	icon_state = "pea"
+	projectile_type = /obj/projectile/bullet/pea
+
+/obj/item/ammo_casing/pea/attack_self(mob/user)
+	qdel(src)
+	var/obj/item/food/grown/peas/peas = new (user.drop_location())
+	user.put_in_hands(peas)
+	to_chat(user, span_notice("You crush the pea in your hand, and it explodes into a small bundle of edible peas."))
```

Comment:
Also, effects after qdel icky:

- Drop `src` to the ground
- Put the new peas in hand + the to_hat
- THEN delete `src`